core/domain/feedback_services.py

```diff
@@ -111,19 +111,27 @@ def create_message(
     # We do a put() even if the status and subject are not updated, so that the
     # last_updated time of the thread reflects the last time a message was
     # added to it.
+    old_status = thread.status
     if message_id != 0 and (updated_status or updated_subject):
         if updated_status and updated_status != thread.status:
             thread.status = updated_status
         if updated_subject and updated_subject != thread.subject:
             thread.subject = updated_subject
+    new_status = thread.status
     thread.put()
 
-    if (user_services.is_user_registered(author_id) and len(text) > 0 and
+    if (user_services.is_user_registered(author_id) and
             feconf.CAN_SEND_EMAILS and
             feconf.CAN_SEND_FEEDBACK_MESSAGE_EMAILS):
             # send feedback message email if user is registered.
-        add_message_to_email_buffer(
-            author_id, exploration_id, thread_id, message_id)
+        if old_status == new_status:
```

Comment:
Don't branch here; send the old status and new status to add_message_to_email_buffer(). That function can check if there's a difference and act accordingly; it's simpler to have all that logic in one place.